common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts

```diff
@@ -102,46 +102,53 @@ export default class LDMLKeyboardXMLSourceFileReader {
   }
 
   private resolveImports(obj: any, subtag: string) {
-    // first, the implied imports
+    // These are in reverse order, because the imports insert at the beginning of the array.
+    // first, the explicit imports
+    for (const asImport of ([...obj['import'] as LKImport[]].reverse())) {
+      this.resolveOneImport(obj, subtag, asImport);
+    }
+    // then, the implied imports
     if (subtag === 'keys') {
       // <import base="cldr" path="techpreview/keys-Latn-implied.xml"/>
       this.resolveOneImport(obj, subtag, {
         base: constants.cldr_import_base,
         path: constants.cldr_implied_keys_import
       });
     }
-
-    // now, the explicit imports
-    for (const asImport of (obj['import'] as LKImport[])) {
-      this.resolveOneImport(obj, subtag, asImport);
-    }
   }
 
   private resolveOneImport(obj: any, subtag: string, asImport: LKImport) {
-    if (asImport.base !== constants.cldr_import_base) {
-      // TODO-LDML: use custom error class CompilerImportResolveError
-      throw new Error(`import element with base ${asImport.base} is unsupported, only ${constants.cldr_import_base} is supported.`);
+    const { base, path } = asImport;
+    if (base !== constants.cldr_import_base) {
+      this.callbacks.reportMessage(CommonTypesMessages.Error_ImportInvalidBase({base, path, subtag}));
+      return;
```

Comment:
We should probably have a `return false;` for failure cases?

Reply:
done